### DISCUSSION TITLE
SDXL Sheduler

### DISCUSSION THREAD

**Discussion by kodinkod** [huggingface/diffusers#7184] [Q&A]
Why in LoRa train scripts we use DDPMScheduler? In original repository sdxl-base in huggingface propose use EulerDiscreteScheduler.

**Comment by sayakpaul** [comment]
The `add_noise()` step and the training objective don't differ much from scheduler to scheduler. That's why. 

We recently added support for performing EDM-style training as well: https://github.com/huggingface/diffusers/pull/7126.

**Reply by kodinkod** [reply]
If I just replace DDMP with EulerDiscret, then the model will be trained incorrectly. Perhaps the fact is that for `EulerDiscret` we have to call scale_model_input, which will work incorrectly because during training the step changes randomly at each step, which is not taken into account in `EulerDiscret .scale_model_input`

**Reply by sayakpaul** [reply]
You're not just replacing the DDPM scheduler, we also added relevant training logic to accommodate the change of the scheduler. Please do consider that too.

**Reply by kodinkod** [reply]
Why are there no conflicts and the model continues to work normally, even if it was trained with one sheduler and used with another?

**Reply by sayakpaul** [reply]
We are performing inference with the same EulerDiscrete scheduler when the `do_edm_style_training` is True. Please look into the script more carefully.

**Reply by kodinkod** [reply]
Could you give an intuition of which part of this table EulerDiscretSheduler implements. How are the columns related? Why are Euler and DDPM in the same column? Are we reducing the ODE solution to a probabilistic model?

I will be very grateful for any response!

<img width="879" alt="Снимок экрана 2024-03-03 в 13 40 31" src="https://github.com/huggingface/diffusers/assets/69761539/405a553c-7146-44af-839c-a1deee9c0023">

paper: https://arxiv.org/pdf/2206.00364.pdf

**Comment by kodinkod** [comment]
> We are performing inference with the same EulerDiscrete scheduler when the `do_edm_style_training` is True. Please look into the script more carefully.

I understand. I meant that when I used the old script without `do_edm_style_training`. The trained model worked correctly with `EulerDiscretSheduler` on inference with my custom lora weights.
